### PR TITLE
basic redux implementation for login form added

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
Let's add some redux stuff...

- Whole Login process is done via redux+redux-thunk
- Request on movies is still called inside component - actual TODO for attendees
- Adding new orders and removing them is additional task as well

### CODE REVIEW

File: src/components/LoginDialog.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      </Modal.Actions>
+    </Modal>
+  );
+};
+
+LoginDialog.propTypes = {
+  onClose: PropTypes.func.isRequired
+};
+
+export default LoginDialog;
```
Comment (Aprillion) [inline src/components/LoginDialog.js]: no newline

File: src/actions/login.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const showLogin = () => ({
+  type: 'SHOW_LOGIN'
+});
+
+export const closeLogin = () => ({
+  type: 'CLOSE_LOGIN'
+});
```
Comment (Aprillion) [inline src/actions/login.js]: github is complaining about no newline at the end of file

File: src/containers/App.js [hunk 3]
```diff
@@ -72,4 +69,17 @@ class App extends Component {
...
+
+const mapDispatchToProps = {
+  onLoginShow: showLogin,
+  onCloseLogin: closeLogin
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(App);
```
Comment (Aprillion) [inline src/containers/App.js]: missing propTypes 
+ App.test.js is missing the new required props

File: src/reducers/rootReducer.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import userIdentity from './userIdentity';
+
+const rootReducer = combineReducers({
+  userIdentity
+});
+
+export default rootReducer;
```
Comment (Aprillion) [inline src/reducers/rootReducer.js]: new line

File: src/reducers/userIdentity.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const userIdentity = (state = null, action) => {
+  return state;
+};
+
+export default  userIdentity;
```
Comment (Aprillion) [inline src/reducers/userIdentity.js]: new line

File: src/reducers/userIdentity.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const userIdentity = (state = null, action) => {
+  return state;
```
Comment (Aprillion) [inline src/reducers/userIdentity.js]: what is the purpose of this reducer that ignores action? missing TODO?

**Review by Aprillion — DISMISSED** [review]

File: src/reducers/rootReducer.js (hunk 4 above)
Comment (jaroslav-kubicek) [reply src/reducers/rootReducer.js]: should be fixed, I also added editorconfig

**Comment by PavelVanecek** [comment]
Do we want to include the immutable bug and explain it? The current reducer is so simple it is not vulnerable but I think the bug + fix should be demonstrated.

**Comment by jaroslav-kubicek** [comment]
I'm finished with this... :)

**Comment by PavelGavlik** [comment]
@PavelVanecek Maybe next time

**Pull request merged**